ccf_transaction_processor/transaction_processor/pdo_tp.cpp

```diff
@@ -0,0 +1,624 @@
+                }
+
+                auto contract_code_hash = contract_r.value().contract_code_hash;
+                auto encoded_code_hash = b64_from_raw(contract_code_hash.data(), contract_code_hash.size());
+
+                string doc_to_sign;
+                doc_to_sign = in.contract_id;
+                doc_to_sign += encoded_code_hash;
+                doc_to_sign += contract_r.value().contract_creator_verifying_key_PEM;
+                for (auto pservice_id : contract_r.value().provisioning_service_ids) {
+                    doc_to_sign += pservice_id;
+                }
+                doc_to_sign += enclave_info_string;
+                auto signature = TPHandler::sign_document(doc_to_sign);
+
+                return make_success(Verify_contract::Out{in.contract_id, encoded_code_hash, \
+                contract_r.value().contract_creator_verifying_key_PEM, contract_r.value().provisioning_service_ids, \
+                enclave_info_string, signature});
+            }
+            return make_error(jsonrpc::StandardErrorCodes::INVALID_PARAMS, "Contract not found");
+
+        };
+
+        //======================================================================================================
+        // get current state info handler implementation
+        auto get_current_state_info_for_contract = [this](Store::Tx& tx, const nlohmann::json& params) {
+
+            const auto in = params.get<Get_current_state_info::In>();
+            auto view = tx.get_view(contracttable);
+            auto contract_r = view->get_globally_committed(in.contract_id);
+
+            if (!contract_r.has_value())
+            {
+                return make_error(jsonrpc::StandardErrorCodes::INVALID_PARAMS, "Contract not found");
+            }
+
+            if(contract_r.value().is_active && contract_r.value().current_state_hash.size() == 0) {
+                return make_error(jsonrpc::StandardErrorCodes::INVALID_PARAMS, "Contract not yet initialized");
+            }
+
+            if (ledger_signer_local == NULL) {
+                return make_error(
+                    jsonrpc::StandardErrorCodes::INVALID_PARAMS, "Cannot find signer locally");
+            }
+
+            auto current_state_hash = contract_r.value().current_state_hash;
+            auto encoded_state_hash = b64_from_raw(current_state_hash.data(), current_state_hash.size());
+            string doc_to_sign = in.contract_id + encoded_state_hash;
+            auto signature = TPHandler::sign_document(doc_to_sign);
+
+            return make_success(Get_current_state_info::Out{encoded_state_hash, contract_r.value().is_active, signature});
+
+        };
+
+        //======================================================================================================
+        // get state details handler implementation
+        auto get_details_about_state = [this](Store::Tx& tx, const nlohmann::json& params) {
```

Comment:
I think a different name is used in sawtooth. It would be good to maintain some consistency since the TP should be roughly equivalent.

Reply:
thank you for the comment. I agree the name is different, and perhaps not ideal. The sawtooth names make little sense either. I am going to stick to this for now.